### PR TITLE
[many ports] add GnuPG libraries

### PR DESCRIPTION
**Pull request by wrobelda**
Adds GnuPG libraries ports 

- Which triplets are supported/not supported? Have you updated the CI baseline?

Currently tested on macOS and Linux only with gettext package installed. Windows builds fail due to missing `autopoint` tool, which is part of gettext. See #13518 for details. 

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by LilyWangL** [comment]
@wrobelda Thanks for your PR. Now `libgpg-error` build failed with the following error:
```
autoreconf: Entering directory `.'
autoreconf: running: autopoint --force
Can't exec "autopoint": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 345.
autoreconf: failed to run autopoint: No such file or directory
autoreconf: autopoint is needed because this package uses Gettext
```

**Comment by wrobelda** [comment]
> @wrobelda Thanks for your PR. Now `libgpg-error` build failed with the following error:

Yes, I explained in the PR why this is happening:

"Windows builds fail due to missing autopoint tool, which is part of gettext. See #13518 for details."

**Comment by wrobelda** [comment]
@LilyWangL I suppose we can release it as-is with your change (91e71d6)

**Comment by Hoikas** [comment]
Per the [maintainer's guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#manifests-and-control-files):
> When adding a new port, use the new manifest syntax for defining a port

**Comment by LilyWangL** [comment]
@wrobelda I test features of `gpgme`, `gpgme[qt]` and `gpgme[python]` both build failed on Linux. Could you fix them? I found if you want install `gpgme[python]`, you need install `swig` and `python3-dev`. But after installing them, `gpgme[python]` build failed with the new error.

**Comment by LilyWangL** [comment]
Pinging @wrobelda for response.

**Comment by wrobelda** [comment]
@LilyWangL the problem with Qt build is due to two issues with vcpkg and/or Qt port: https://github.com/microsoft/vcpkg/issues/14225

Another issue is identical to this to https://github.com/microsoft/vcpkg/issues/13649#issuecomment-699286758. In this particular case, `./configure` attempts to locate Qt tools:

```
configure:19433: checking for moc
configure:19463: result: no
configure:19482: checking moc version
configure:19487: result: no
configure:19533: checking for moc-qt5
configure:19563: result: no
configure:19628: checking for qtchooser
configure:19658: result: no
``
However, there are no parameters that let me override those paths explicitly. The tools are available at `vcpkg_installed/x64-osx/tools/qt5/bin/`, but this location is not added to `PATH` variable. What's the solution here? I truly don't understand why aren't all of these locations added by default?

**Comment by wrobelda** [comment]
@LilyWangL the problem with Qt build is due to two issues with vcpkg and/or Qt port: https://github.com/microsoft/vcpkg/issues/14225

Another one is identical to this to https://github.com/microsoft/vcpkg/issues/13649#issuecomment-699286758 – in this case, `./configure` attempts to locate qt tools:

```
configure:19433: checking for moc
configure:19463: result: no
configure:19482: checking moc version
configure:19487: result: no
configure:19533: checking for moc-qt5
configure:19563: result: no
configure:19628: checking for qtchooser
configure:19658: result: no
```
However, the gpgme's `configure` does not offer parameters that would let me explicitly provide paths to those executables. They are available at `vcpkg_installed/x64-osx/tools/qt5/bin/`, but this location is not added to `$PATH`. What's the solution here? Why aren't these locations added to $PATH by default anyway? This makes the whole proces unnecessarily tedious.

**Comment by wrobelda** [comment]
@LilyWangL I disabled Qt for now until the #14225 is fixed. Python bindings should be fixed now, too.

File: ports/gpgme/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    PATCHES
+       b05747aea79083106b27ac7d829db3e65e192682.patch
+       disable-docs.patch
+)
+
+list(REMOVE_ITEM FEATURES core)
+string(REPLACE ";" "," LANGUAGES "${FEATURES}")
+
+if("python" IN_LIST FEATURES)
+    vcpkg_find_acquire_program(PYTHON3)
```
Comment (Hoikas) [inline ports/gpgme/portfile.cmake:18]: This will cause python bindings to be built for the host instead of the target.

Comment (wrobelda) [reply ports/gpgme/portfile.cmake:18]: It works on Windows, so I assumed it uses the vcpkg's own package as expected. What is the correct way of doing it, then?

Comment (Hoikas) [reply ports/gpgme/portfile.cmake:18]: I'm not sure the specifics behind gpgme's build system, but you will want it to link against the python libraries in the vcpkg installed tree. These are usually named things like libpython3.8, libpython3.8_d, python38, and python38_d - depending on the platform.

Comment (wrobelda) [reply ports/gpgme/portfile.cmake:18]: @Hoikas I am quite sure GpgMe only needs the Python interpreter and SWIG to generate Python bindings; libpython would be needed if they actually interacted with Python from C code which I am almost certain they don't.

Comment (wrobelda) [reply ports/gpgme/portfile.cmake:18]: I disabled Python bindings for now. The library works as is, let's get it merged and revisit at a later time.

File: ports/gpgme/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,35 @@
+vcpkg_fail_port_install(MESSAGE "${PORT} currently only supports unix platform" ON_TARGET "Windows")
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO gpg/gpgme
+    REF gpgme-1.14.0
+    SHA512 b4608fd1d9a4122d8886917274e323afc9a30494c13a3dea51e17e9779f925bf8d67e584434d6a13018f274a6cbcf0a5e36f2fea794a065906bbb556b765398e
+    HEAD_REF master
+    PATCHES
+       b05747aea79083106b27ac7d829db3e65e192682.patch
```
Comment (PhoebeHui) [inline ports/gpgme/portfile.cmake:10]: Could you update the patch name to a more readable name?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
Thanks for the PR!

Except the patch name, others looks good for me. thanks for your contribution!

**Comment by wrobelda** [comment]
> Except the patch name, others looks good for me. thanks for your contribution!

Thank you as well. Fixed the patch name, it's ready to go!

**Review by PhoebeHui — APPROVED** [review]

File: ports/gpgme/CONTROL [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Source: gpgme
+Version: 1.14.0
+Homepage: https://gnupg.org/software/gpgme/index.html
+Description: A library designed to make access to GnuPG easier for applications
+Build-Depends: libgpg-error, libassuan
+Default-Features: cpp
+Supports: linux|osx
```
Comment (strega-nil) [inline ports/gpgme/CONTROL:7]: Probably best to do

```suggestion
Supports: !windows

```
?

File: ports/libassuan/CONTROL [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Source: libassuan
+Version: 2.5.3
+Homepage: https://gnupg.org/software/libassuan/index.html
+Description: A library implementing the so-called Assuan protocol
+Build-Depends: libgpg-error
+Supports: linux|osx
```
Comment (strega-nil) [inline ports/libassuan/CONTROL:6]:
```suggestion
Supports: !windows

```

same thing here (and for the rest, don't feel like adding them)

File: ports/libassuan/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        --disable-silent-rules
+        --with-libgpg-error-prefix=${CURRENT_INSTALLED_DIR}/tools/libgpg-error
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig() 
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(INSTALL ${SOURCE_PATH}/COPYING.LIB DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (strega-nil) [inline ports/libassuan/portfile.cmake:27]:
```suggestion
file(INSTALL ${SOURCE_PATH}/COPYING.LIB DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

```

File: ports/libgcrypt/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        --disable-silent-rules
+        --with-libgpg-error-prefix=${CURRENT_INSTALLED_DIR}/tools/libgpg-error
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig() 
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(INSTALL ${SOURCE_PATH}/COPYING.LIB DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (strega-nil) [inline ports/libgcrypt/portfile.cmake:27]:
```suggestion
file(INSTALL ${SOURCE_PATH}/COPYING.LIB DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

```

File: ports/libgpg-error/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        --disable-doc
+        --disable-silent-rules
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig() 
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(INSTALL ${SOURCE_PATH}/COPYING.LIB DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (strega-nil) [inline ports/libgpg-error/portfile.cmake:25]:
```suggestion
file(INSTALL ${SOURCE_PATH}/COPYING.LIB DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

```

**Review by strega-nil — COMMENTED** [review]
Minor things, then I'll merge :)

**Comment by PhoebeHui** [comment]
@wrobelda, could you address the review request?

**Comment by wrobelda** [comment]
> @wrobelda, could you address the review request?

I did, just that I am without a workstation temporarily. Will address these suggestions asap. Thank you!

**Comment by wrobelda** [comment]
@strega-nil Suggestions applied. My apologies for the delay.

File: ports/gpgme/CONTROL [hunk 8]
```diff
@@ -0,0 +1,11 @@
+Source: gpgme
+Version: 1.14.0
+Homepage: https://gnupg.org/software/gpgme/index.html
+Description: A library designed to make access to GnuPG easier for applications
+Build-Depends: libgpg-error, libassuan
+Default-Features: cpp
+Supports: !windows
+
+
```
Comment (PhoebeHui) [inline ports/gpgme/CONTROL:9]:
```suggestion
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks a bunch!

**Pull request merged**